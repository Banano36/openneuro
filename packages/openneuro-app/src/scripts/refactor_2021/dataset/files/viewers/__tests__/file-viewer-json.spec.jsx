import React from 'react'
import { shallow } from 'enzyme'
import { FileViewerJsonRaw } from '../file-viewer-json.jsx'

describe('File Viewer - JSON', () => {
  it('renders with valid JSON', () => {
    const validJson = JSON.stringify({ thing: 1, name: 'string' })
    const wrapper = shallow(<FileViewerJsonRaw jsonRaw={validJson} />)
    expect(wrapper).toMatchSnapshot()
  })
  it('renders with invalid JSON', () => {
    const wrapper = shallow(<FileViewerJsonRaw jsonRaw="1234;" />)
    expect(wrapper).toMatchSnapshot()
  })
})
