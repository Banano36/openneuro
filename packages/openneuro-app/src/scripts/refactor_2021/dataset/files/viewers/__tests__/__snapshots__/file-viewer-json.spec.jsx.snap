// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`File Viewer - JSON renders with invalid JSON 1`] = `
<div
  className="container"
>
  <div
    className="grid"
  >
    <div
      className="col col-lg"
    >
      <h3>
        Tree
      </h3>
      <hr />
      1234;
    </div>
    <div
      className="col"
    >
      <h3>
        Raw
      </h3>
      <hr />
      <Styled(pre)>
        1234;
      </Styled(pre)>
    </div>
  </div>
</div>
`;

exports[`File Viewer - JSON renders with valid JSON 1`] = `
<div
  className="container"
>
  <div
    className="grid"
  >
    <div
      className="col col-lg"
    >
      <h3>
        Tree
      </h3>
      <hr />
      {"thing":1,"name":"string"}
    </div>
    <div
      className="col"
    >
      <h3>
        Raw
      </h3>
      <hr />
      <Styled(pre)>
        {"thing":1,"name":"string"}
      </Styled(pre)>
    </div>
  </div>
</div>
`;
